code/modules/mob/living/simple_animal/hostile/giant_spider.dm

```diff
@@ -136,6 +136,23 @@
 	poison_per_bite = 5
 	move_to_delay = 5
 
+//variant of the hunter granted as the hos's pet, it has no poison but deals more damage to other spiders and heals when attacking them.
+/mob/living/simple_animal/hostile/poison/giant_spider/hunter/hos
+	name = "Commander Chorizopes"
+	desc = "An extremely rare variant of spider found one day chomping on Araneus, He prefers spider meat moreso than human meat so we kept him around in Araneus's place."
+	icon_state = "hunter"
+	icon_living = "hunter"
+	icon_dead = "hunter_dead"
+	poison_per_bite = 0
+	move_to_delay = 5
+
+/mob/living/simple_animal/hostile/poison/giant_spider/hunter/hos/AttackingTarget()
+	. = ..()
+	if(. && isliving(target) && /mob/living/simple_animal/hostile/poison/giant_spider)

```

Comment:
does `&& /mob/living/simple_animal/hostile/poison/giant_spider)` just work or what? where's the is_type()

Reply:
no, it doesn't work, what you need to do is
`if(. && istype(target, /mob/living/simple_animal/hostile/poison/giant_spider))` and typecast accordingly

Reply:
ok cool i thought i was just going crazy

Reply:
in the end, i was the only one going crazy